### ISSUE TITLE
Investigate why listRepositories doesn't work with HDFS

### ISSUE THREAD

**Issue by smacker** [src-d/gemini#32]
Most probably something is wrong with documentation.

**Comment by smacker** [comment]
There is a configuration for hadoop which lives in https://github.com/src-d/baremetal/blob/master/contrib/spark/core-site.xml
You can see it in spark master container and spark on cluster loads it correctly.

In gemini we create `SparkSession` passing that master.
After that we obtain `sparkContext`.
And sparkContext has property `hadoopConfiguration` which we can use passing it to `FileSystem.get`.
The problem is it doesn’t work :joy:

And here is why. While hadoop allows defining default fs, we don’t do it as you can see in the file above. And by default, it’s `fs.defaultFS = file:///`
Then if we look at hadoop fs API we will see 3 methods `get`:
- get(Configuration conf)
- get(URI uri, Configuration conf)
- get(URI uri, Configuration conf, String user)

Right now we use the first one. https://github.com/src-d/gemini/blob/4ed375cb20a226344bf9c2a89d64c55435dfcbc5/src/main/scala/tech/sourced/gemini/HashSparkApp.scala#L74

It returns default fs (`file:///`). That’s why when we try to list path `hdfs://...` it raises exception `Wrong FS`.
But with a second one it will create FS for passed url. And path will be found.

Engine v0.2 had the same problem: https://github.com/src-d/engine/pull/245/files
And berserker doesn’t have: https://github.com/src-d/berserker/blob/cfc38e6159ab94922a263aa504110f29c837d2ea/src/main/scala/tech/sourced/berserker/FsUtils.scala#L78

In conclusion to make `listRepositories` work in gemini we just need to use `FileSystem.get(uri, conf)` instead of `FileSystem.get(conf)`
